### PR TITLE
[Release] Add heuristic on how to manually handle the changelog

### PR DESCRIPTION
**Pull request by cipolleschi**
This PR improves the contributing section, listing a set of heuristic we follow to adjust the changelog when we are preparing a new release of React Native.

### CODE REVIEW

File: website/contributing/release-branch-cut-and-rc0.md [hunk 1]
```diff
@@ -94,6 +94,21 @@ npx @rnx-kit/rn-changelog-generator --base v0.68.2 --compare v0.69.0-rc.0 \
...
+Aat the end of the generated changelog, there could be two special sections: the **Failed to Parse** section and the **Uknown** section.
+
+The **Failed to Parse** section contains commits that has a `## Changelog:` entry in their summary but, due to typos or other problems, the tool was not able to parse and automatically attribute them to the right section.
+
+The **Unknown** section is populated with commit that landed without the `## Changelog:` entry in the summary.
+
+For both these categories, we have to manually go through the listed commits and move them to the right section, based on the actual change they introduce. The following are a set of heuristic we follow to manually update the changelog:
+
+- Commit of internal changes can be **deleted** from the Changelog.
+  Examples: commit on BUCK files or code refactoring that do not change behaviors or interfaces
```
Comment (kelset) [inline website/contributing/release-branch-cut-and-rc0.md:108]:
```suggestion
  - Examples: commit on BUCK files or code refactoring that do not change behaviours or interfaces
```
Assuming you want to mirror the other bullet point, you want a sub bullet here too

File: website/contributing/release-branch-cut-and-rc0.md [hunk 2]
```diff
@@ -94,6 +94,21 @@ npx @rnx-kit/rn-changelog-generator --base v0.68.2 --compare v0.69.0-rc.0 \
 
 Create a pull request of this change to `react-native` repo and add the `Changelog` label.
 
+#### Manually Adjust the Changelog
+
+At the end of the generated Changelog, there could be two additional sections: the **Failed to Parse** section and the **Uknown** section.
```
Comment (iBotPeaches) [inline website/contributing/release-branch-cut-and-rc0.md:99]: `Unknown` :)

File: website/contributing/release-branch-cut-and-rc0.md [hunk 3]
```diff
@@ -94,6 +94,21 @@ npx @rnx-kit/rn-changelog-generator --base v0.68.2 --compare v0.69.0-rc.0 \
...
+
+The **Failed to Parse** section contains commits that has a `## Changelog:` entry in their summary but, due to typos or other problems, the tool was not able to parse and automatically attribute them to the right section.
+
+The **Unknown** section is populated with commit that landed without the `## Changelog:` entry in the summary.
+
+For both these categories, we have to manually go through the listed commits and move them to the right section, based on the actual change they introduce. The following are a set of heuristic we follow to manually update the changelog:
+
+- Commit of internal changes can be **deleted** from the Changelog.
+  - Examples: commit on BUCK files or code refactoring that do not change behaviours or interfaces
+- Commit which bumps depemendencies should be **moved to the `Changed`** section
```
Comment (iBotPeaches) [inline website/contributing/release-branch-cut-and-rc0.md:109]: `dependencies`

File: website/contributing/release-branch-cut-and-rc0.md (hunk 2 above)
Comment (cipolleschi) [reply website/contributing/release-branch-cut-and-rc0.md]: Thanks for spotting this.

File: website/contributing/release-branch-cut-and-rc0.md (hunk 3 above)
Comment (cipolleschi) [reply website/contributing/release-branch-cut-and-rc0.md]: Thanks for spotting this.

**Pull request merged**